internal/rest/middleware.go

```diff
@@ -24,25 +35,61 @@ func appendMiddlewares(mw ...middleware) func(http.Handler) http.Handler {
 // logger logs the request and response
 func logger(next http.Handler) http.Handler {
 	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
-		if r.URL.Path != "/v2/healthz" { // reduce log spam
-			slog.Debug("received request", "method", r.Method, "path", r.URL.Path)
+		if slices.Contains(ignoredPaths, r.URL.Path) {
+			next.ServeHTTP(w, r)
+			return
 		}
+
+		slog.Debug("received request", "method", r.Method, "path", r.URL.Path)
 		next.ServeHTTP(w, r)
 	})
 }
 
+// requestDurationMetrics measures the duration of the request and records it for Prometheus
+func requestDurationMetrics(next http.Handler) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		if slices.Contains(ignoredPaths, r.URL.Path) {
+			next.ServeHTTP(w, r)
+			return
+		}
+
+		start := time.Now()
+		next.ServeHTTP(w, r)
+		d := time.Since(start).Seconds()
+		metrics.ResponseTime.Observe(d)
+	})
+}
+
+// responseCounterMetrics counts the number of responses and records it for Prometheus
```

Comment:
and this the throughput, i.e., QPS? the same question about testing

Reply:
No, this metric labels requests with endpoint, method and response code - it then increments a counter for every request with matching labels.

`GET /v2/clusters 200" would be a common request, whereas `POST /v2/clusters 200` would be cluster creation.

This could come in handy when scale testing if more and more requests for cluster creation start failing, this metric can be graphed over time.